go/heif/heif.go

```diff
@@ -871,56 +872,56 @@ func imageFromYCbCr(i *image.YCbCr) (*Image, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to add Y plane: %v", err)
 	}
-	pY.SetData([]byte(i.Y), i.YStride)
+	pY.setData([]byte(i.Y), i.YStride)
 
 	// TODO: Might need to be updated for other SubsampleRatio values.
 	halfW, halfH := (w+1)/2, (h+1)/2
 	pCb, err := out.NewPlane(ChannelCb, halfW, halfH, depth)
 	if err != nil {
 		return nil, fmt.Errorf("failed to add Cb plane: %v", err)
 	}
-	pCb.SetData([]byte(i.Cb), i.CStride)
+	pCb.setData([]byte(i.Cb), i.CStride)
 	pCr, err := out.NewPlane(ChannelCr, halfW, halfH, depth)
 	if err != nil {
 		return nil, fmt.Errorf("failed to add Cr plane: %v", err)
 	}
-	pCr.SetData([]byte(i.Cr), i.CStride)
+	pCr.setData([]byte(i.Cr), i.CStride)
 
 	return out, nil
 }
 
-func EncodeFromImage(img image.Image, compression Compression, quality int, lossless losslessMode, logging loggingLevel) (*Context, error) {
+func EncodeFromImage(img image.Image, compression Compression, quality int, lossless LosslessMode, logging LoggingLevel) (*Context, error) {
 	var out *Image
 
-	switch cm := img.ColorModel(); cm {
+	switch t := img.(type) {
```

Comment:
This could even be simplified more:
```go
switch i := img.(type) {
case *image.NRGBA:
  // i has now type *image.NRGBA
case *image.Gray:
  // i has now type *image.NRGBA
...
}
```

You don't need to cast inside the `case`, see https://tour.golang.org/methods/16 for details.

Reply:
TIL. Thanks, much better now :)